backend/groth16/bn254/icicle/icicle.go

```diff
@@ -202,33 +267,44 @@ func Prove(r1cs *cs.R1CS, pk *ProvingKey, fullWitness witness.Witness, opts ...b
 	wireValues := []fr.Element(solution.W)
 
 	start := time.Now()
+	poks := make([]curve.G1Affine, len(pk.CommitmentKeys))
 
+	for i := range pk.CommitmentKeys {
+		var err error
+		if poks[i], err = pk.CommitmentKeys[i].ProveKnowledge(privateCommittedValues[i]); err != nil {
```

Comment:
In essence this is a big MSM, in this PR we still do it in CPU.

For big circuits which use field emulation this MSM can be huge (hundreds of thousands of inputs), so should be worthwile to move to ICICLE as well?